File: EcoBookAiBackend/src/main/java/com/ecobook/dto/SolicitacaoMaterialDTO.java
package com.ecobook.dto;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public class SolicitacaoMaterialDTO {
    private String id;
    private String titulo;
    private String descricao;
    private String imagemUrl;
    private String disciplina;
    private String nivelEnsino;
    private Integer ano;
    private String status;
    private String cidade;
    private String bairro;
    private String doadorNome;
}

File: EcoBookAiAndroid/app/src/main/kotlin/com/ecobook/dto/SolicitacaoMaterialDTO.kt
package com.ecobook.dto

data class SolicitacaoMaterialDTO(
    val id: String,
    val titulo: String,
    val descricao: String? = null,
    val imagemUrl: String? = null,
    val disciplina: String,
    val nivelEnsino: String,
    val ano: Int? = null,
    val status: String,
    val cidade: String,
    val bairro: String,
    val doadorNome: String,
    val doadorFotoPerfilUrl: String? = null
)
